x-pack/filebeat/input/cel/input.go

```diff
@@ -251,12 +253,21 @@ func (i input) run(env v2.Context, src *source, cursor map[string]interface{}, p
 			log.Debugw("request state", logp.Namespace("cel"), "state", redactor{state: state, cfg: cfg.Redact})
 			metrics.executions.Add(1)
 			start := i.now().In(time.UTC)
-			state, err = evalWith(ctx, prg, ast, state, start)
+			state, err = evalWith(ctx, prg, ast, state, start, wantDump)
 			log.Debugw("response state", logp.Namespace("cel"), "state", redactor{state: state, cfg: cfg.Redact})
 			if err != nil {
+				var dump dumpError
 				switch {
 				case errors.Is(err, context.Canceled), errors.Is(err, context.DeadlineExceeded):
 					return err
+				case errors.As(err, &dump):
+					dir := strings.ReplaceAll(cfg.FailureDumpPath, "*", sanitizeFileName(env.IDWithoutName))
+					path := filepath.Join(dir, "dump_"+i.now().In(time.UTC).Format("20060102150405.999Z"))
```

Comment:
Should we be using the [`lumberjackTimestamp`](https://github.com/elastic/beats/blob/main/x-pack/filebeat/input/cel/input.go#L728) format for consistency with tracer file names?

Reply:
I need to think about that; the base will be different, so the clean-up code for tracing won't see them anyway. But more significantly, the logic that's used for clean-up of traces is not appropriate for clean-up of failure dumps since traces are used for a period and so should be removed when the period is over, but failure dumps are intended to be used one-shot (maybe I need to reconsider that).

Reply:
The deletion logic is implemented. Another thing that we should consider is whether we cap the number of dumps. This is done in the trace request logging, but that is done by the lumberjack package. The query I guess is how rapidly we expect the filesystem to get filled; this should only happen once per periodic evaluation, so we can just ask users to set this to a value that is long enough for them to collect the dump and then turn off the dump collection.

Reply:
Thanks for changing significantly to cover this. Added another review. 
Only docs change required.